src/test/java/com/microsoft/sqlserver/jdbc/datatypes/DataTypesTest.java

```diff
@@ -1670,4 +1672,134 @@ public void testUpdateMisc() throws Exception {
             }
         }
     }
+    
+    /*
+     * SQL Server compatibility level >= 130 (SQL Server 2016 and after) doesn't round up or down equally for
+     * datetime. The rounding logic below represents the actual rounding logic employed by datetime
+     * columns when determining their last millisecond digit.
+     */
+    @Test
+    public void testDateTimeInsertUpdate() throws Exception {
+        assumeTrue(!isSqlAzureDW(), TestResource.getResource("R_cursorAzureDW"));

```

Comment:
Can this test be made compatible for Azure DW too?

Reply:
I tried, but the cursor type needs to be updatable, which means it's not supported for DW.